crates/goose-server/src/state.rs

```diff
@@ -21,15 +38,34 @@ impl AppState {
     pub fn new(provider_config: ProviderConfig, secret_key: String) -> Result<Self> {
         let provider = factory::get_provider(provider_config.clone())?;
         let mut agent = Agent::new(provider);
+
+        dbg!("Adding DeveloperSystem");
         agent.add_system(Box::new(DeveloperSystem::new()));
 
+        // Only add NonDeveloperSystem if we're not in a git repository and don't have a .goosehints file
+        let in_git = is_in_git_repository();
+        let has_hints = has_goosehints_file();
+
+        if !in_git && !has_hints {
```

Comment:
For most users initially, this will be turned on by default since the UI starts up in the home directory. Depending on the user, that may not be the best experience (?). Def need a way to add/select systems on the fly, but for now maybe a configuration in the menu can enable this?

Reply:
yeah - actually can add menu and also make it detect just git installed at all (doesn't seem to affect developer experience, but then you don't want extra in the prompt you don't need!)

Reply:
changed it to use a menu